packages/foundation-models/package.json

```diff
@@ -22,13 +22,14 @@
   "scripts": {
     "compile": "tsc",
     "compile:cjs": "tsc -p tsconfig.cjs.json",
-    "test": "NODE_OPTIONS=--experimental-vm-modules jest",
+    "test": "pnpm generate-zod && NODE_OPTIONS=--experimental-vm-modules jest",
     "lint": "eslint \"**/*.ts\" && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -c",
     "lint:fix": "eslint \"**/*.ts\" --fix && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -w --log-level error",
     "check:public-api": "node --loader ts-node/esm ../../scripts/check-public-api-cli.ts",
     "generate:azure-openai": "openapi-generator --generateESM --clearOutputDir -i ./src/azure-openai/spec/inference.yaml -o ./src/azure-openai/client --schemaPrefix AzureOpenAi",
     "postgenerate:azure-openai": "rm ./src/azure-openai/client/inference/*.ts && pnpm lint:fix",
-    "generate-zod": "ts-to-zod src/azure-openai/azure-openai-embedding-types.ts src/azure-openai/ts-to-zod/azure-openai-embedding-types.zod.ts"
+    "generate-zod-config": "node --loader ts-node/esm ../../scripts/generate-zod-config.ts './src/azure-openai/client/inference/schema/*.ts' ./src/azure-openai/ts-to-zod",
+    "generate-zod": "pnpm generate-zod-config && ts-to-zod --all --skipValidation"
```

Comment:
[pp] I want to emphasize that this is totally my personal opinion only: I feel that the zod check is not suuuper important, but quite prominent here. I also think that the zod test is not a normal unit test, so I would prefer to move this to a separate package under tests so that when just running unit tests we don't need to generate this every time. Also I think the separation would be cleaner with this.

other opinions @SAP/ai-sdk-js-team?

Reply:
I quite like this idea.

Reply:
Also, the config should not be regenerated everytime (only when we update the spec and therefore the types)

Reply:
I will try to move the schema tests into a separate package.

Reply:
Refactored.